agent/handlers/task_server_setup_sudo_integ_test.go

```diff
@@ -0,0 +1,427 @@
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os/exec"
+	"testing"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	mock_dockerstate "github.com/aws/amazon-ecs-agent/agent/engine/dockerstate/mocks"
+	agentV4 "github.com/aws/amazon-ecs-agent/agent/handlers/v4"
+	mock_stats "github.com/aws/amazon-ecs-agent/agent/stats/mock"
+	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
+	mock_ecs "github.com/aws/amazon-ecs-agent/ecs-agent/api/ecs/mocks"
+	apitaskstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	ni "github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+	faulttype "github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/fault/v1/types"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/utils/execwrapper"
+
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	endpointId               = "endpoint"
+	taskARN                  = "t1"
+	containerID              = "cid"
+	containerName            = "sleepy"
+	imageName                = "busybox"
+	imageID                  = "bUsYbOx"
+	cpu                      = 1024
+	memory                   = 512
+	containerPort            = 80
+	associationType          = "elastic-inference"
+	associationName          = "dev1"
+	associationEncoding      = "base64"
+	associationValue         = "val"
+	family                   = "sleep"
+	version                  = "1"
+	eniIPv4Address           = "10.0.0.2"
+	macAddress               = "06:96:9a:ce:a6:ce"
+	privateDNSName           = "ip-172-31-47-69.us-west-2.compute.internal"
+	subnetGatewayIpv4Address = "172.31.32.1/20"
+	taskCredentialsID        = "taskCredentialsId"
+	defaultIfname            = "eth0"
+	clusterName              = "default"
+	availabilityzone         = "us-west-2b"
+	vpcID                    = "test-vpc-id"
+	containerInstanceArn     = "containerInstanceArn-test"
+	startEndpoint            = "/api/%s/fault/v1/%s/start"
+	stopEndpoint             = "/api/%s/fault/v1/%s/stop"
+	statusEndpoint           = "/api/%s/fault/v1/%s/status"
+	defaultNamespace         = "host"
+	requestMaxRetries        = 3
+	requestTimeoutInSeconds  = 5 * time.Second
+
+	port               = 4321
+	protocol           = "tcp"
+	trafficType        = "ingress"
+	delayMilliseconds  = 100
+	jitterMilliseconds = 200
+	lossPercent        = 6
+
+	serverPort = 3333
+)
+
+var (
```

Comment:
nit and non-blocking: Using getter functions over `var`s is better for long term maintenance since that prevents tests from modifying the originally defined values. Once we have a test that modifies these variables they cannot reliably be used by other tests as constants.